performance.lic

```diff
@@ -0,0 +1,153 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#performance
+=end
+
+custom_require.call(%w(common drinfomon spellmonitor))
+  
+class Performance
+  
+  include DRC
+  
+  def initialize
+    no_pause_all
+    startTime = Time.new.to_i
+
+    @settings = get_settings
+    @song_list = get_data('perform').perform_options
+    UserVars.song = @song_list.first.first unless UserVars.song
+    fput('release ecry') if DRSpells.active_spells["Eillie's Cry"].to_i > 0
+
+    instrument = @settings.instrument
+
+    hand_armor = @settings.hand_armor
+    if instrument
+      bput("get my #{instrument}",'You get')
+      bput("remove my #{hand_armor}",'You pull off','Remove what')
+      bput("stow my #{hand_armor}",'You put','Stow what')
+    end
+    
+    if play_song
+      loop do 
+        runTime = Time.new.to_i - startTime
+        break if runTime > 600
+        break if done_training
+        line = get
+        pause 0.05 unless line
+        case line
+        when /You begin (.*)/, /You continue (.*)/, /You find yourself relaxing (.*)/, /You concentrate on your sense of rhythm (.*)/, /You continue (.*)/
+          if done_training
+            exit
+          end
+        when /You finish (.*)/
+          if not done_training
```

Comment:
💡 By convention, we've been using `! && ||` rather than `not and or`.

Another style would be `unless done_training?`.

Probably my most preferred option would be to reverse the order of the `then` and `else` so you don't need a `!` in the first place:
```
if done_training?
  exit
else
  play_song
end
```